pandas/_libs/reduction.pyx

```diff
@@ -66,9 +66,7 @@ cdef class _BaseGrouper:
             object.__setattr__(cached_ityp, '_index_data', islider.buf)
             cached_ityp._engine.clear_mapping()
             cached_ityp._cache.clear()  # e.g. inferred_freq must go
-            object.__setattr__(cached_typ._mgr._block, 'values', vslider.buf)
-            object.__setattr__(cached_typ._mgr._block, 'mgr_locs',
```

Comment:
might get a small boost by setting _mgr_locs to BlockPlacement(slice(...)) instead of going through the mgr_locs property